w3-blob.md

```diff
@@ -304,7 +350,7 @@ The `args.blob` field MUST be set to the `Blob` the space is allocated for.
 
 #### Allocation Cause
 
-The `args.cause` field MUST be set to the [Link] for an [Add Blob] task, that caused an allocation.
+The `args.cause` field MUST be set to the [Link] for the [Add Blob] task, that caused an allocation.
 
 ### Allocate Blob Receipt
 
```

Comment:
See below. Should BlobPut be the following in terms of capability definition:

```ts
type BlobPut = {
  cmd: "/http/put"
  sub: DID
  args: {
    url: [".out.ok.address.url", Link<BlobAlloc>]
    headers: [".out.ok.address.headers", Link<BlobAlloc>]
    body: Blob
  }
  meta: {
    keys: {[key: DID]: bytes}
  }
}
```

Reply:
We talked about this in sync, but capturing the summary here

1. General idea is that arguments are defined in terms of types.
2. Argument can be literals or they can be `ucan/await`  selectors, in which case they effectively act as `await` in js and assume that selector will lead to the value of the expected type.
    - If awaited task fails /  does not have a selected path errors propagate

Unfortunately currently our schema definitions and specifically `.invoke` method on them is unable to recognize `ucan/await` pattern, so short term we could define schema more explicitly using something like

```js
const Await = Schema.struct({
  'ucan/await': Schema.tuple([Schema.text(),  Schema.link()])
})

const Put = Schema.struct({
  can: "http/put",
  with: Schema.did(),
  nb: Schema.struct({
    url: Schema.text().or(Await),
    headers: Schema.dictionary({ value: Schema.string() }).or(Await),
    body: blob
  })
})
```